### PR TITLE
Fix `sort_values` simplification for `{head|tail}`

### PR DESCRIPTION
**Pull request by hendrikmakait**
For single partitions, `sort_values` supports sorting by multiple columns with each column sorted ascending or descending independently. When these values are not always the same, this cannot be simplified to `smallest`. Since the simplification does not save us much with a single partition, I avoid simplifying at all in this case.

### CODE REVIEW

File: dask_expr/tests/test_shuffle.py [hunk 1]
```diff
@@ -372,6 +372,67 @@ def test_sort_head_nlargest(df):
...
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values(["x", "y"], ascending=[True]).head(10, compute=False).expr
+    b = df.nsmallest(10, columns=["x", "y"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    with pytest.raises(
+        NotImplementedError,
+        match="Dask currently only supports a single boolean for ascending",
+    ):
```
Comment (fjetter) [inline dask_expr/tests/test_shuffle.py:394]: funny you should mention that... https://github.com/dask/dask/pull/10864

File: dask_expr/tests/test_shuffle.py [hunk 2]
```diff
@@ -372,6 +372,83 @@ def test_sort_head_nlargest(df):
...
+    result = df.sort_values("x", ascending=[True]).head(10)
+    expected = pdf.sort_values("x", ascending=[True]).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values(["x"], ascending=[True]).head(10)
+    expected = pdf.sort_values(["x"], ascending=[True]).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values(["x"], ascending=[False]).head(10)
+    expected = pdf.sort_values(["x"], ascending=[False]).head(10)
```
Comment (hendrikmakait) [inline dask_expr/tests/test_shuffle.py:435]: This case also fails in the current implementation, which is why https://github.com/coiled/benchmarks/pull/1290/files#diff-ed9173e9cde3ccc498e25c8485d7c2060041c6b7cb0228f26e1c16603a3f3b10L478 works as a fix.

**Comment by hendrikmakait** [comment]
CI failures are unrelated.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**